src/planner/ostree.rs

```diff
@@ -0,0 +1,406 @@
+            "/etc/systemd/system/nix.mount",
+            None,
+            None,
+            0o0644,
+            create_bind_mount_buf,
+            false,
+        )
+        .await
+        .map_err(PlannerError::Action)?;
+        plan.push(create_bind_mount_unit.boxed());
+
+        let ensure_symlinked_units_resolve_buf = "\
+        [Unit]\n\
+        Description=Ensure Nix related units which are symlinked resolve\n\
+        After=nix.mount\n\
+        Requires=nix.mount\n\
+        DefaultDependencies=no\n\
+        \n\
+        [Service]\n\
+        Type=oneshot\n\
+        RemainAfterExit=yes\n\
+        ExecStart=/usr/bin/systemctl daemon-reload\n\
+        ExecStart=/usr/bin/systemctl restart --no-block nix-daemon.socket\n\
+        \n\
+        [Install]\n\
+        WantedBy=sysinit.target\n\
+    "
+        .to_string();
+        let ensure_symlinked_units_resolve_unit = CreateFile::plan(
+            "/etc/systemd/system/ensure-symlinked-units-resolve.service",
+            None,
+            None,
+            0o0644,
+            ensure_symlinked_units_resolve_buf,
+            false,
+        )
+        .await
+        .map_err(PlannerError::Action)?;
+        plan.push(ensure_symlinked_units_resolve_unit.boxed());
+
+        // We need to remove this path since it's part of the read-only install.
+        let mut shell_profile_locations = ShellProfileLocations::default();
+        if let Some(index) = shell_profile_locations
+            .fish
+            .vendor_confd_prefixes
+            .iter()
+            .position(|v| *v == PathBuf::from("/usr/share/fish/"))
+        {
+            shell_profile_locations
+                .fish
+                .vendor_confd_prefixes
+                .remove(index);
+        }
+
+        plan.push(
+            StartSystemdUnit::plan("nix.mount".to_string(), false)
+                .await
+                .map_err(PlannerError::Action)?
+                .boxed(),
+        );
+
+        plan.push(
+            ProvisionNix::plan(&self.settings.clone())
+                .await
+                .map_err(PlannerError::Action)?
+                .boxed(),
+        );
+        plan.push(
+            CreateUsersAndGroups::plan(self.settings.clone())
+                .await
+                .map_err(PlannerError::Action)?
+                .boxed(),
+        );
+        plan.push(
+            ConfigureNix::plan(shell_profile_locations, &self.settings)
+                .await
+                .map_err(PlannerError::Action)?
+                .boxed(),
+        );
+
+        if has_selinux {
+            plan.push(
+                ProvisionSelinux::plan("/etc/nix-installer/selinux/packages/nix.pp".into())
+                    .await
+                    .map_err(PlannerError::Action)?
+                    .boxed(),
+            );
+        }
+
+        plan.push(
+            ConfigureInitService::plan(self.init.init, self.init.start_daemon)
```

Comment:
As I previously noted I do not think `--init none` is valid for this planner so this would be

```suggestion
            ConfigureInitService::plan(InitSystem::Systemd, true)
```